#ifndef BOOK_H
#define BOOK_H
#include <QWidget>
#include <QString>
#include <QKeyEvent>
#include <QEvent>
#include <QMainWindow>
#include "variables.h"
namespace Ui {
class book;
}

class book : public QWidget
{
    Q_OBJECT
signals:
       void refresh();

public:
    explicit book(QWidget *parent = nullptr);
    ~book();
    void eachpage();
    void keyPressEvent(QKeyEvent *event) override;
    void show_book();
    int calc_damage(int monster_id);
    void fun(int page);
    int book_a;
    int book_b;
    int book_tem = 0;
    int id[40];
    int damage[3];
    int Totalpage = 1;
    int page = 1;
    QImage mo[62];
    character* braver;
    monster m_array[MONSTER_NUM] =
    {   monster(35,19,2,0,1,0, "绿色史莱姆", "51"),
      //hp,at,df,gold,exp,id,"name","img";
        monster(45,23,3,1,1,1, "红色史莱姆","52"),
        monster(70,45,15,6,1,2,"黑色史莱姆","53"),
        monster(400,149,94,18,2,3,"史莱姆王","54"),
        monster(35,30,3,2,1,4,"小蝙蝠","55"),
        monster(70,61,29,8,1,5,"大蝙蝠","56"),
        monster(200,201,119,25,3,6,"红蝙蝠","57"),
        monster(70,80,0,5,1,7,"骷髅","58"),
        monster(320,189,15,17,2,8,"骷髅剑士","59"),
        monster(210,140,102,15,2,9,"骷髅兵","60"),
        monster(500,330,34,30,4,10,"骷髅队长","61"),
        monster(20,50,36,4,1,11,"初级守卫","62"),
        monster(210,120,118,20,3,12,"中级守卫","63"),
        monster(310,724,700,54,6,13,"高级守卫","64"),
        monster(10,70,110,12,2,14,"石头怪","65"),
        monster(110,90,30,8,1,15,"丧尸","66"),
        monster(535,305,180,41,5,16,"丧尸剑士","67"),
        monster(180,123,81,14,2,17,"双手剑士","68"),
        monster(920,1330,660,73,8,18,"剑王","69"),
        monster(300,188,150,22,3,19,"八爪鱼","70"),
        monster(50,18,4,3,1,20,"水术士","71"),
        monster(160,30,37,10,2,21,"雷术士","72"),
        monster(330,90,170,38,4,22,"炎术士","73"),
        monster(496,112,182,43,5,23,"大法师","74"),
        monster(700,120,250,46,5,24,"暗黑魔法师","75"),
        monster(868,155,880,84,8,25,"极地法师","76"),
        monster(850,175,490,63,7,26,"死灵大法师","77"),
        monster(905,200,830,89,9,27,"死灵术士","78"),
        monster(1130,220,540,67,7,28,"火炎大法师","79"),
        monster(610,120,210,44,5,29,"魔法师首领","80"),
        monster(810,155,502,57,6,30,"魔法师元帅","81"),
        monster(1620,210,730,82,9,31,"魔王","82"),
        monster(1330,250,800,92,10,32,"混沌术士","83"),
        monster(3900,310,1104,700,300,33,"魔神·些多","84"),
        monster(402,556,245,51,5,34,"毒蝙蝠","85"),
        monster(660,500,387,48,5,35,"毒丧尸","86"),
        monster(938,530,400,52,5,36,"毒八爪鱼","87"),
        monster(3500,1446,999,800,190,37,"毒龙","88"),
        monster(830,570,390,20,3,38,"骷髅战士","89"),
        monster(999,999,90,55,6,39,"骷髅头目","90"),
        monster(1144,999,826,74,8,40,"闪光八爪鱼","91"),
        monster(1100,1240,800,22,3,41,"不死战士","92"),
        monster(210,113,47,12,2,42,"骑士","93"),
        monster(370,233,130,27,3,43,"铁骑士","94"),
        monster(1002,1000,592,73,8,44,"近卫骑士","95"),
        monster(2000,1600,980,800,190,45,"剑神·沙士","96"),
        monster(176,98,66,12,2,46,"史莱姆人","97"),
        monster(280,234,164,36,4,47,"四手史莱姆人","98"),
        monster(888,661,552,59,6,48,"水银人","99"),
        monster(800,640,320,30,4,49,"重生骷髅兵","100"),
        monster(1097,1024,800,80,8,50,"烈焰丧尸","101"),
        monster(1400,1400,601,33,4,51,"不死骷髅兵","102"),
        monster(1100,1120,900,22,3,52,"死灵战士","103"),
        monster(20,780,800,57,6,53,"铁怪","104"),
        monster(1500,370,190,400,70,54,"魔王·格勒第","105"),
        monster(3000,1530,1530,1800,180,55,"黄金龙","106"),
        monster(4400,1287,1200,800,190,56,"斗神·高巴","107"),
        monster(2020,1080,700,400,100,57,"魔界八爪鱼","108"),
        monster(1220,1010,633,70,8,58,"魔王近卫兵","109"),
        monster(1080,1011,603,72,8,59,"冻死骨","110"),
        monster(932,721,562,61,6,60,"邪眼史莱姆","111"),
        monster(20,1399,999,99,9,61,"鬼邪石","112")
    };


private:
    Ui::book *ui;
};


#endif // BOOK_H
